src/parser.ts

```diff
@@ -187,9 +183,197 @@ function getFunctionName(func: ts.FunctionLikeDeclaration): string {
             }
         }
     }
+    return appendNecessaryPrefix(nameText, func);
+}
+
+function appendNecessaryPrefix(
+    funcName: string,
+    func: ts.FunctionLikeDeclaration
+) {
+    let nameText = funcName;
+    let propValue: ts.FunctionLikeDeclaration | ts.ClassLikeDeclaration = func;
+
+    // prepend class prefix if function is inside a class
+    const classPrefix = getClassPrefix(func);
+    if (classPrefix) {
+        nameText =
+            nameText === "" ? classPrefix : `${classPrefix}.${nameText}`;
+        const classPropValue = getParentClassProperty(func);
+        if (classPropValue) {
+            // if the class is a property of an object literal, we want to
+            // walk the object literal chain of the class instead of the function
+            propValue = classPropValue;
+        } else {
+            // this means the class is not inside an object literal and we are done
+            return nameText;
+        }
+    }
+
+    // prepend object literal prefix if function (or enclosing class)
+    // is inside an object literal
+    const objectLiteralPrefix = getObjectLiteralPrefix(propValue);
+    if (objectLiteralPrefix) {
+        nameText = `${objectLiteralPrefix}.${nameText}`;
+    }
     return nameText;
 }
 
+function getMemberNodeForFunction(
```

Comment:
This function is doing two things. Also I think "Property" in this context means "is it a class instance field declaration?" 

Can we split out `IsMember` from `IsClassField`?

Reply:
It's not just class instance field though, it can also be a property in an object literal, e.g.
```javascript
let a = {
    b: function() {}
}
```
I am using the word "Property" to be consistent with the API's naming - our check boils down to checking `ts.isPropertyDeclaration(func.parent) || ts.isPropertyAssignment(func.parent)`. It felt like `isProperty` is the common one between the two, but I am happy to hear other alternatives.